metering-billing/lambdas/src/main/java/com/amazon/aws/partners/saasfactory/metering/onboarding/SubscriptionService.java

```diff
@@ -24,49 +25,78 @@
+import com.stripe.model.Product;
+import com.stripe.model.ProductCollection;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import java.util.*;
-import java.util.stream.Collectors;
-import java.util.stream.Stream;
 
 public class SubscriptionService {
-    private final static Map<String, String> CORS = Stream
-            .of(new AbstractMap.SimpleEntry<String, String>("Access-Control-Allow-Origin", "*"))
-            .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
-    private final static Logger LOGGER = LoggerFactory.getLogger(SubscriptionService.class);
+    private static final Map<String, String> CORS = Map.of("Access-Control-Allow-Origin", "*");
+    private static final Logger LOGGER = LoggerFactory.getLogger(SubscriptionService.class);
+    private static final String API_GATEWAY_HOST = System.getenv("API_GATEWAY_HOST");
+    private static final String API_GATEWAY_STAGE = System.getenv("API_GATEWAY_STAGE");
+    private static final String API_TRUST_ROLE = System.getenv("API_TRUST_ROLE");
 
     public SubscriptionService() {
         LOGGER.info("Version Info: " + Utils.version(this.getClass()));
+        if (Utils.isBlank(API_GATEWAY_HOST)) {
+            throw new IllegalStateException("Missing required environment variable API_GATEWAY_HOST");
+        }
+        if (Utils.isBlank(API_GATEWAY_STAGE)) {
+            throw new IllegalStateException("Missing required environment variable API_GATEWAY_STAGE");
+        }
+        if (Utils.isBlank(API_TRUST_ROLE)) {
+            throw new IllegalStateException("Missing required environment variable API_TRUST_ROLE");
+        }
     }
-    public APIGatewayProxyResponseEvent getPlans(Map<String, Object> event, Context context) throws JsonProcessingException {
+
+    public APIGatewayProxyResponseEvent getPlans(Map<String, Object> event, Context context) {
         if (Utils.warmup(event)) {
-            //LOGGER.info("Warming up");
             return new APIGatewayProxyResponseEvent().withHeaders(CORS).withStatusCode(200);
         }
 
-        long startTimeMillis = System.currentTimeMillis();
-        LOGGER.info("SubscriptionService::getPlans starting");
+        final long startTimeMillis = System.currentTimeMillis();
+        Utils.logRequestEvent(event);
 
-        // create an array of key-value pairs
-        //ObjectMapper MAPPER = new ObjectMapper();
+        APIGatewayProxyResponseEvent response;
 
-        //Utils.logRequestEvent(event);
-        APIGatewayProxyResponseEvent response = null;
-        ArrayNode plans = JsonNodeFactory.instance.arrayNode();
-        for (SubscriptionPlan plan : SubscriptionPlan.values()) {
-            ObjectNode planN = JsonNodeFactory.instance.objectNode();
-            planN.put("planId", plan.name());
-            planN.put("planName", plan.getLabel());
-            plans.add(planN);
+        Stripe.apiKey = BillingUtils.getBillingApiKey(API_GATEWAY_HOST, API_GATEWAY_STAGE, API_TRUST_ROLE);
```

Comment:
Note that this can throw an `java.util.NoSuchElementException: No value present` if app config doesn't have a billing api key or if app config simply hasn't been saved yet. A better client experience would be to return either an empty list or a `404` when no plans exist. However, there is no way to invoke this API through the admin web app until you've saved the app config, so impact right now is very minimal.